- Buildeploy.targets: creation of nuget packages not only by providin…

a-powolozki opened the pull request:
Allow to create nuget packages not only by providing a nuspec file but also by providing a project file. This automaticaly includes all dependent packages into the incorporated nuspec file and all references, which could not be resolved as nuget packages will be included as referenced assemblies.

Commit:
- Buildeploy.targets: creation of nuget packages not only by providing a nuspec file but also by providing a csproj file.

Commit:
Line ending setting for target files

Commit:
-Buildeploy.targets: converted line endings to crlf

a-powolozki commented:
Hope the line endings are now ok.

a-powolozki committed:
- Tasks/ResolveRedistAssemblies.cs: use RelativeDir metadata item on resolving assemblies paths for redistribution

a-powolozki committed:
Possibility to omit default excludes during NugetPackage Task

a-powolozki committed:
Project file to build and package buildeploy from scratch updated to avoid the chicken or the egg problem

a-powolozki committed:
NugetPackage Taks: rename property containing PackageId to Id as this token name is the original token name used by nuget for package identifier

derjabkin commented on Tasks/NugetPackage.cs:
```diff
@@ -37,14 +37,18 @@ public class NugetPackage : PackageTaskBase
             var archive = new NugetArchive(NugetExePath, Path.GetFullPath(NuspecFile), OutputDirectory, Version, Log);
             if (!string.IsNullOrWhiteSpace(PackageId))
-                archive.AddProperty("PackageId", PackageId);
+                archive.AddProperty("Id", PackageId);
```
This would break existing projects. Please add also the PackageId property.

a-powolozki committed:
Restoring backward compatibility on setting NuGet PackageId property

derjabkin commented on Buildeploy.targets:
```diff
@@ -1,408 +1,417 @@
-            PushLocation="%(OctoConfig.PushLocation)"
-            ApiKey="%(OctoConfig.ApiKey)"
-            />
-    <Copy SourceFiles="$(VersionRedistRoot)\ClickOnce\$(ClickOnceFilePrefix)%(OctoConfig.Identity)_$(Version).zip"
```
Why you deleted the copy commands?

a-powolozki replied:
Deleted copy commands are in my opinion copy / pasted commands which are exactly the same as used inside ClickOnce target. That's why i thought, their presence in the OctoPackage Target is a mistake, causing files to be copied two times, if they would be generated.

derjabkin replied:
OctoPackage is not related to ClickOnce.

On Sat, Jun 20, 2015 at 7:46 PM Alexander Powolozki <
notifications@github.com> wrote:

> In Buildeploy.targets
> https://github.com/CMSSudhaus/buildeploy/pull/1#discussion_r32886411:
> 
> > -            MinimumRequiredVersion="%(OctoConfig.MinimumRequiredVersion)"
> > -            RequireLatestVersion="%(OctoConfig.RequireLatestVersion)"
> > -            Version="$(Version)"
> > -            TargetFramework="$(ClickOnceTargetFramework)"
> > -            CreateDesktopShortcut="%(OctoConfig.CreateDesktopShortcut)"
> > -            IconFile="%(OctoConfig.IconFile)"
> > -            WebsiteBasePath="$(CompiledSite)"
> > -            CombineWithWebsite="%(OctoConfig.CombineWithWebsite)"
> > -            WebsiteFiles="@(CompiledSiteWithConfig)"
> > -                 LinkAssembliesWithManifestAsFile="$(ClickOnceLinkAssembliesWithManifestAsFile)"
> > -            SkipClickOnce="%(OctoConfig.SkipClickOnce)"
> > -            PackageId="%(OctoConfig.PackageId)"
> > -            PushLocation="%(OctoConfig.PushLocation)"
> > -            ApiKey="%(OctoConfig.ApiKey)"
> > -            />
> > -    <Copy SourceFiles="$(VersionRedistRoot)\ClickOnce\$(ClickOnceFilePrefix)%(OctoConfig.Identity)_$(Version).zip"
> 
> Deleted copy commands are in my opinion copy / pasted commands which are
> exactly the same as ised during ClickOnce job. That's why i thought, their
> presence in the OctoPackage Target is a mistake, causing files to be copied
> two times, if they would be generated.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CMSSudhaus/buildeploy/pull/1/files#r32886411.

a-powolozki replied:
Exactly, you are absolutly right!

The copy comands do deploy generated clickonce packages. This deployment is absolutely wrong place insde the OctoPackage target and is already done inside the ClickOnce target.

derjabkin replied:
And When the ClickOnce Target never invoked? Who will copy
ServerRedistFiles?

On Sat, Jun 20, 2015 at 7:59 PM Alexander Powolozki <
notifications@github.com> wrote:

> In Buildeploy.targets
> https://github.com/CMSSudhaus/buildeploy/pull/1#discussion_r32886518:
> 
> > -            MinimumRequiredVersion="%(OctoConfig.MinimumRequiredVersion)"
> > -            RequireLatestVersion="%(OctoConfig.RequireLatestVersion)"
> > -            Version="$(Version)"
> > -            TargetFramework="$(ClickOnceTargetFramework)"
> > -            CreateDesktopShortcut="%(OctoConfig.CreateDesktopShortcut)"
> > -            IconFile="%(OctoConfig.IconFile)"
> > -            WebsiteBasePath="$(CompiledSite)"
> > -            CombineWithWebsite="%(OctoConfig.CombineWithWebsite)"
> > -            WebsiteFiles="@(CompiledSiteWithConfig)"
> > -                 LinkAssembliesWithManifestAsFile="$(ClickOnceLinkAssembliesWithManifestAsFile)"
> > -            SkipClickOnce="%(OctoConfig.SkipClickOnce)"
> > -            PackageId="%(OctoConfig.PackageId)"
> > -            PushLocation="%(OctoConfig.PushLocation)"
> > -            ApiKey="%(OctoConfig.ApiKey)"
> > -            />
> > -    <Copy SourceFiles="$(VersionRedistRoot)\ClickOnce\$(ClickOnceFilePrefix)%(OctoConfig.Identity)_$(Version).zip"
> 
> Exactly, you are absolutly right!
> 
> The copy comands do deploy generated clickonce packages. This deployment
> is absolutely wrong place insde the OctoPackage target and is already done
> inside the ClickOnce target.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CMSSudhaus/buildeploy/pull/1/files#r32886518.

a-powolozki replied:
I think this should be extracted into separate Target which should be invoked in the same manner like ClickOnce and OctoPackage.

derjabkin commented on Buildeploy.targets:
```diff
@@ -1,408 +1,417 @@
+
+		<Error Condition="$(RedistRoot)==''" Text="RedistRoot or CustomRedistRoot properties are not specified."/>
+
+		<!--
```
Please, Remove commented out code

a-powolozki committed:
- Buildeploy.targets: removed not needed comments

a-powolozki committed:
- Refactoring of properties generation to use it by multiple targets
- Refactoring of preparation and redistribution of binaries to be able to add and check additional binaries by probject

a-powolozki committed:
Merge branch 'master' of https://github.com/a-powolozki/buildeploy
